### ISSUE TITLE
Add teleportOrientation(?) for rigidbody

### ISSUE THREAD

**Issue by daredevildave** [playcanvas/engine#782]
Currently `rigidbody.teleport` requires you to set a position if you want to change the orientation. We should  provide a method if you only want to change the orientation.

**Comment by vkalpias** [comment]
We could add a few methods like `setPosition`, `setRotation` on the `rigidbody`.

**Comment by qiangzi7723** [comment]
Yes,I think so!That will be a greater way to change the orientation in the future.

**Comment by Maksims** [comment]
Perhaps setPosition and setRotation on rigidbody might confuse as being same name as on Entity. Different naming might help to distinguish behaviour and relevance to entiyty or rigidbody.

**Comment by willeastcott** [comment]
The 'nullification' (is that a word) of setPosition/Rotation when adding a dynamic rigid body component has been a source of confusion and endless support queries since we implemented the API. Can we make look again at whether we can solve _that_ problem? The reason we introduced this was because we didn't want the entity to know about the rigidbody component. But perhaps we should look at this again...

**Comment by Maksims** [comment]
@willeastcott you mean to make setPosition and setRotation on Entity, to actually change rigidbody instead? Perhaps that would dramatically simplify the whole world of working with physical objects, indeed.

**Comment by willeastcott** [comment]
I was thinking more that updating the transform of the entity via the standard functions could set some flag which the rigidbody could query and then handle. So the entity still wouldn't need to know about the rigidbody component. But I'm sure I would have thought about this before so there's probably a problem with this.

**Comment by Maksims** [comment]
That probably wont work, as setPosition and setRotation, and other methods should have immediate effect, rather than delayed.

**Comment by willeastcott** [comment]
Do it with events? The rigidbody component could register an event on the entity if the position or orientation is updated and respond accordingly?

**Comment by Maksims** [comment]
Could, but feels like an overhead. I would rather check if entity has method _syncTransform, and if does, it would call it every time transform has been changed.
Then rigidbody would define that method when component is added and would remove it when removed.

That way entity does not need to know about rigidbody, only about possible _syncTransform. Overhead is minimal here - single if to check if method is defined.
Events would have too much of an overhead to deal with, as transform changes - is very-very often thing in update loop.

**Comment by willeastcott** [comment]
Yeah, that'd work (for now). It's internal so we can always change it later if necessary.
